### ISSUE TITLE
Crash when multiple functions have the same structure pointer argument

### ISSUE THREAD

**Issue by wanglin86769** [Koromix/koffi#115]
Hello,
Today I encountered a problem, that is when multiple functions have the same structure pointer argument, it will crash when the pointer is passed to the second function and use the pointer in C++.

The following is the snippet I was using to reproduce the problem.

C++ code:
```
typedef struct PVDef {
    char *name;
    char *type;
    int count;
} PVDef;

extern "C" {
    epicsShareFunc int epicsShareAPI testfunc1(PVDef *pvs, int count);
    epicsShareFunc int epicsShareAPI testfunc2(PVDef *pvs, int count);
} 

int displayPVDef(PVDef *pv) {
    std::cout << "name: " << pv->name << std::endl;
    std::cout << "type: " << pv->type << std::endl;
    std::cout << "count: " << pv->count << std::endl;
    return 0;
}

int testfunc1(PVDef *pvs, int count) {
    PVDef *pv;
    for(int i = 0; i < count; i++) {
        pv = pvs + i;
        displayPVDef(pv);
    }
    return 0;
}

int testfunc2(PVDef *pvs, int count) {
    PVDef *pv;
    for(int i = 0; i < count; i++) {
        pv = pvs + i;
        displayPVDef(pv);
    }
    return 0;
}
```

JavaScript code:
```
const PVDef = koffi.struct('PVDef', {
    name: 'char *',
    type: 'char *',
    count: 'int'
});

const testfunc1 = libpcas.func('testfunc1', 'int', ['PVDef *', 'int']);
const testfunc2 = libpcas.func('testfunc2', 'int', ['PVDef *', 'int']);

let pv001 = {
    name: 'test:name001',
    type: 'int',
    count: 1
}

let pv002 = {
    name: 'test:name002',
    type: 'float',
    count: 10
}

let pvList = [pv001, pv002];
// testfunc1(pvList, pvList.length);
testfunc2(pvList, pvList.length);

setInterval(() => {
    console.log("setInterval()");
}, 1000);
```

testfunc1() is the first function and always execute successfully. However, testfunc2 will crash at "pv->name" in displayPVDef().

The are three ways to make testfunc2() succeed,

1. Switch the function declaration order in JavaScript like this,
```
const testfunc2 = libpcas.func('testfunc2', 'int', ['PVDef *', 'int']);
const testfunc1 = libpcas.func('testfunc1', 'int', ['PVDef *', 'int']);
```

2. Comment the first function in JavaScript like this,
```
// const testfunc1 = libpcas.func('testfunc1', 'int', ['PVDef *', 'int']);
const testfunc2 = libpcas.func('testfunc2', 'int', ['PVDef *', 'int']);
```

3. Change the argument of the first function to non-PVDef type like this,
```
const testfunc1 = libpcas.func('testfunc1', 'int', ['int *', 'int']);
const testfunc2 = libpcas.func('testfunc2', 'int', ['PVDef *', 'int']);
```

I am not sure if it can be reproduced in other environments, my environment is Windows 10, Node.js 16.19.0 and Visual Studio 2022.

**Comment by Koromix** [comment]
I can reproduce here, on Linux, what a weird one :) Checking it tonight!

**Comment by Koromix** [comment]
This was a regression in Koffi 2.6.11.

I've published [Koffi 2.6.12](https://koffi.dev/changelog#koffi-2-6-12-2023-12-11) which should fix this issue.
